zerver/lib/actions.py

```diff
@@ -2426,6 +2432,19 @@ def gather_subscriptions_helper(user_profile):
         else:
             unsubscribed.append(stream_dict)
 
+    never_subscribed_stream_ids = [item for item in all_streams_id if item not in sub_unsub_stream_ids]
+    for stream_id in never_subscribed_stream_ids:
+        stream = Stream.objects.get(id=stream_id)
```

Comment:
You don't need this query as well -- you can look up the stream inside of `stream_dicts` once you make the modification above